test/hca-rjsf/config/veteranInformation.unit.spec.jsx

```diff
@@ -2,36 +2,107 @@ import React from 'react';
 import { findDOMNode } from 'react-dom';
 import { expect } from 'chai';
 import ReactTestUtils from 'react-dom/test-utils';
+import sinon from 'sinon';
 
-import { DefinitionTester } from '../../util/schemaform-utils.jsx';
+import { DefinitionTester, submitForm } from '../../util/schemaform-utils.jsx';
 import formConfig from '../../../src/js/hca-rjsf/config/form.js';
 
 describe('HCA veteranInformation', () => {
   it('should render veteranInformation page', () => {
+    const onSubmit = sinon.spy();
     const { schema, uiSchema } = formConfig.chapters.veteranInformation.pages.veteranInformation;
     const form = ReactTestUtils.renderIntoDocument(
       <DefinitionTester
           schema={schema}
           data={{}}
+          onSubmit={onSubmit}
           uiSchema={uiSchema}/>
     );
     const formDOM = findDOMNode(form);
 
     expect(formDOM.querySelectorAll('input, select').length).to.equal(5);
     expect(formDOM.querySelector('#root_veteranFullName_first')).not.to.be.null;
+
+    submitForm(form);
+
+    expect(formDOM.querySelectorAll('.usa-input-error').length).to.equal(2);
+    expect(onSubmit.called).to.be.false;
   });
 
   it('should render birthInformation page', () => {
+    const onSubmit = sinon.spy();
     const { schema, uiSchema } = formConfig.chapters.veteranInformation.pages.birthInformation;
     const form = ReactTestUtils.renderIntoDocument(
       <DefinitionTester
           schema={schema}
           data={{}}
+          onSubmit={onSubmit}
           uiSchema={uiSchema}/>
     );
     const formDOM = findDOMNode(form);
 
     expect(formDOM.querySelectorAll('input, select').length).to.equal(6);
     expect(formDOM.querySelector('#root_veteranSocialSecurityNumber')).not.to.be.null;
+
+    submitForm(form);
+
+    expect(formDOM.querySelectorAll('.usa-input-error').length).to.equal(2);
+    expect(onSubmit.called).to.be.false;
+  });
+
+  it('should render demographicInformation page', () => {
+    const onSubmit = sinon.spy();
+    const { schema, uiSchema } = formConfig.chapters.veteranInformation.pages.demographicInformation;
+    const form = ReactTestUtils.renderIntoDocument(
+      <DefinitionTester
+          schema={schema}
+          data={{}}
+          onSubmit={onSubmit}
+          uiSchema={uiSchema}/>
+    );
+    const formDOM = findDOMNode(form);
+
+    expect(formDOM.querySelectorAll('input, select').length).to.equal(8);
+    expect(formDOM.querySelector('#root_gender')).not.to.be.null;
```

Comment:
Nah, I think it's fine. Just something I hadn't seen on any of the edu forms and was wondering if we should be doing that there.